src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -802,6 +802,62 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input Iceberg INT column. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @return Parsed integer
+   */
+  static int validateAndParseIcebergInt(String columnName, Object input, long insertRowIndex) {
+    if (input instanceof Number) {
+      return ((Number) input).intValue();
```

Comment:
can't this cause an exception if the input were a long value / had a value > Int.maxvalue?

Reply:
Discussed via slack. Added range check and rounding.